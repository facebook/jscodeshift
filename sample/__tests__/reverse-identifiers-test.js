/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * An example of writing a unit test for a jscodeshift script using the
 * `defineTest` helper bundled with jscodeshift. This will run the
 * reverse-identifiers.js transform with the input specified in the
 * reverse-identifiers-input file, and expect the output to be the same as that
 * in reverse-identifiers-output.
 */

'use strict';

jest.autoMockOff();
const defineTest = require('../../src/testUtils').defineTest;
const defineInlineTest = require('../../src/testUtils').defineInlineTest;
const defineSnapshotTestFromFixture = require('../../src/testUtils').defineSnapshotTestFromFixture;
const transform = require('../reverse-identifiers');

defineTest(__dirname, 'reverse-identifiers');

defineTest(__dirname, 'reverse-identifiers', null, 'typescript/reverse-identifiers', { parser: 'ts' });

describe('reverse-identifiers', () => {
  defineInlineTest(transform, {}, `
var firstWord = 'Hello ';
var secondWord = 'world';
var message = firstWord + secondWord;`,`
var droWtsrif = 'Hello ';
var droWdnoces = 'world';
var egassem = droWtsrif + droWdnoces;
  `);
  defineInlineTest(transform, {},
    'function aFunction() {};',
    'function noitcnuFa() {};',
    'Reverses function names'
  );
});

// the snapshot output of this file should be the same as reverse-identifiers.output.js
defineSnapshotTestFromFixture(__dirname, transform, {}, 'reverse-identifiers');
