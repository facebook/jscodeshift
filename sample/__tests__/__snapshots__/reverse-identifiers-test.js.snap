// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transforms correctly 1`] = `
"var droWtsrif = 'Hello ';
var droWdnoces = 'world';
var egassem = droWtsrif + droWdnoces;

class ooF {
  @detaroced
  *rab() { }
}"
`;
